tests/sample_size/test_metrics.py

```diff
@@ -76,6 +80,29 @@ def test_boolean_metric_get_probability_too_small(self):
             "Error: Please provide a float between 0 and 1 for probability.",
         )
 
+    @parameterized.expand(product([True, False], [2, 5], [200]))
+    @patch("sample_size.metrics.BooleanMetric.variance")
+    @patch("scipy.stats.uniform.rvs")
+    @patch("scipy.stats.norm.rvs")
+    def test_boolean_generate_p_value(self, true_null, variants, sample_size, mock_norm, mock_uniform, mock_variance):
+        p_value_generator = mock_uniform if true_null else mock_norm
+        pval = np.zeros(variants - 1)
+        p_value_generator.return_value = pval
+        mock_variance.__get__ = MagicMock(return_value=self.DEFAULT_MOCK_VARIANCE)
+
+        boolean = BooleanMetric(self.DEFAULT_PROBABILITY, self.DEFAULT_MDE)
+        p = boolean.generate_p_value(true_null, sample_size, variants)
+
+        if true_null:
+            mock_uniform.assert_called_once_with(0, 1, size=variants - 1)
+            mock_norm.assert_not_called()
+            assert_array_equal(p, pval)
+        else:
+            effect_sample_size = self.DEFAULT_MDE / float(np.sqrt(2 * self.DEFAULT_MOCK_VARIANCE / sample_size))
+            mock_uniform.assert_not_called()
+            mock_norm.assert_called_once_with(loc=effect_sample_size, size=variants - 1)
+            self.assertEqual(len(p), len(pval))
```

Comment:
I think with some diffrent mocking you can test the value of what the p-value generator returns in both cases
```suggestion
    @parameterized.expand(product([True, False], [2, 5], [200]))
    @patch("sample_size.metrics.BooleanMetric.variance")
    @patch("scipy.stats.uniform.rvs")
    @patch("scipy.stats.norm")
    def test_boolean_generate_p_value(
        self, true_null, variants, sample_size, mock_norm, mock_uniform_rvs, mock_variance
    ):
        mock_norm.rvs.return_value = -ord("🌮")
        p_value_generator = mock_uniform_rvs if true_null else mock_norm.sf
        pval = np.zeros(variants - 1)
        p_value_generator.return_value = pval
        mock_variance.__get__ = MagicMock(return_value=self.DEFAULT_MOCK_VARIANCE)

        boolean = BooleanMetric(self.DEFAULT_PROBABILITY, self.DEFAULT_MDE)
        p = boolean.generate_p_value(true_null, sample_size, variants)

        if true_null:
            mock_uniform_rvs.assert_called_once_with(0, 1, size=variants - 1)
            mock_norm.assert_not_called()
        else:
            effect_sample_size = self.DEFAULT_MDE / float(np.sqrt(2 * self.DEFAULT_MOCK_VARIANCE / sample_size))
            mock_uniform_rvs.assert_not_called()
            mock_norm.rvs.assert_called_once_with(loc=effect_sample_size, size=variants - 1)
            mock_norm.sf.assert_called_once_with(np.abs(mock_norm.rvs.return_value))

        assert_array_equal(p, pval)
```